wallet/common/symbol/src/utils/transaction-from-dto.js

```diff
@@ -0,0 +1,635 @@
+
+	return {
+		...baseTransaction,
+		linkAction: LinkActionMessage[transaction.linkAction],
+		linkedPublicKey,
+		linkedAccountAddress
+	};
+};
+
+export const nodeKeyLinkTransactionFromDTO = (transactionDTO, config) => {
+	const { transaction } = transactionDTO;
+	const baseTransaction = baseTransactionFromDTO(transactionDTO, config);
+	const { linkedPublicKey } = transaction;
+	const linkedAccountAddress = addressFromPublicKey(linkedPublicKey, config.networkProperties.networkIdentifier);
+
+	return {
+		...baseTransaction,
+		linkAction: LinkActionMessage[transaction.linkAction],
+		linkedPublicKey,
+		linkedAccountAddress
+	};
+};
+
+export const accountKeyLinkTransactionFromDTO = (transactionDTO, config) => {
+	const { transaction } = transactionDTO;
+	const baseTransaction = baseTransactionFromDTO(transactionDTO, config);
+	const { linkedPublicKey } = transaction;
+	const linkedAccountAddress = addressFromPublicKey(linkedPublicKey, config.networkProperties.networkIdentifier);
+
+	return {
+		...baseTransaction,
+		linkAction: LinkActionMessage[transaction.linkAction],
+		linkedPublicKey,
+		linkedAccountAddress
+	};
+};
+
+export const getUnresolvedIdsFromTransactionDTOs = transactionDTOs => {
+	const fieldsMap = {
+		[TransactionType.TRANSFER]: {
+			address: ['recipientAddress'],
+			mosaicArray: ['mosaics']
+		},
+		[TransactionType.ADDRESS_ALIAS]: {
+			namespace: ['namespaceId']
+		},
+		[TransactionType.MOSAIC_ALIAS]: {
+			namespace: ['namespaceId']
+		},
+		[TransactionType.MOSAIC_SUPPLY_REVOCATION]: {
+			address: ['sourceAddress'],
+			mosaic: ['mosaicId']
+		},
+		[TransactionType.MULTISIG_ACCOUNT_MODIFICATION]: {
+			addressArray: ['addressAdditions', 'addressDeletions']
+		},
+		[TransactionType.HASH_LOCK]: {
+			mosaic: ['mosaicId']
+		},
+		[TransactionType.SECRET_LOCK]: {
+			address: ['recipientAddress'],
+			mosaic: ['mosaicId']
+		},
+		[TransactionType.SECRET_PROOF]: {
+			address: ['recipientAddress']
+		},
+		[TransactionType.ACCOUNT_ADDRESS_RESTRICTION]: {
+			addressArray: ['restrictionAdditions', 'restrictionDeletions']
+		},
+		[TransactionType.MOSAIC_ADDRESS_RESTRICTION]: {
+			address: ['targetAddress'],
+			mosaic: ['mosaicId']
+		},
+		[TransactionType.MOSAIC_GLOBAL_RESTRICTION]: {
+			mosaic: ['referenceMosaicId']
+		},
+		[TransactionType.ACCOUNT_METADATA]: {
+			address: ['targetAddress']
+		},
+		[TransactionType.MOSAIC_METADATA]: {
+			address: ['targetAddress'],
+			mosaic: ['targetMosaicId']
+		},
+		[TransactionType.NAMESPACE_METADATA]: {
+			address: ['targetAddress'],
+			namespace: ['targetNamespaceId']
+		}
+	};
+
+	const config = {
+		fieldsMap,
+		mapNamespaceId: id => id,
+		mapMosaicId: id => id,
+		mapTransactionType: type => type,
+		getBodyFromTransaction: transaction => transaction.transaction,
+		getHeightFromTransaction: transaction => transaction.meta.height,
+		verifyAddress: address => address.length === 48
+	};
+
+	return getUnresolvedIdsFromTransactions(transactionDTOs, config);
+};
```

Comment:
Just to verify, are these converting from REST objects to Wallet objects?

Reply:
Yes they are.